src/core/killall.c

```diff
@@ -77,14 +77,11 @@ static bool ignore_proc(pid_t pid, bool warn_rootfs) {
         return true;
 }
 
-static void wait_for_children(Set *pids, sigset_t *mask, usec_t timeout) {
+static int wait_for_children(Set *pids, sigset_t *mask, usec_t timeout) {
         usec_t until;
 
         assert(mask);
 
-        if (set_isempty(pids))
-                return;
```

Comment:
hmm, why drop this?

Reply:
There was a reason, but mostly a bad one... Put it back